huxley/api/serializers/user.py

```diff
@@ -7,21 +7,39 @@
 
 from datetime import datetime
 
+from django.core.mail import send_mail
+
 from huxley.accounts.models import User, School
 from huxley.api import validators
 from huxley.api.serializers.school import SchoolSerializer
 
 
 class UserSerializer(ModelSerializer):
-    school = SchoolSerializer(required=False, read_only=True)
+    school = SchoolSerializer(required=False)
 
     class Meta:
         model = User
         fields = ('id', 'username', 'first_name', 'last_name', 'user_type',
                   'school', 'committee',)
-        read_only_fields = ('id', 'username', 'first_name', 'last_name',
-                            'user_type', 'committee',)
+        read_only_fields = ('id', 'username', 'user_type', 'committee',)
 
+    def update(self, instance, validated_data):
+        if validated_data.get('school'):
+            school_data = validated_data.pop('school')
+            School.objects.filter(name=instance.school).update(**school_data)
+            User.objects.filter(username=instance).update(**validated_data)
+            send_mail('{0} has updated its registration information'.format(instance.school),
+                      'New registraion information for {0}: \n\n'.format(instance.school) \
+                      + 'Advisor: \n' \
+                      + '\n'.join(['{0}: {1}'.format(field, validated_data[field]) for field in validated_data]) \
+                      + '\n\nSchool: \n' \
+                      + '\n'.join(['{0}: {1}'.format(field, school_data[field]) for field in school_data]),
+                      'tech@bmun.org',
+                      ['external@bmun.org'], fail_silently=False)
+            return instance
+        else:
+            return super(UserSerializer, self).update(instance, validated_data)
```

Comment:
Oh I see, we only want to update if the school has changed. Would the following work?
```python
def update(self, instance, validated_data):
    result = super(UserSerializer, self).update(instance, validated_data)
    if 'school' in validated_data:
      # send email...
    return result
    
```